README.md

```diff
@@ -31,38 +33,64 @@ const client = new SgidClient({
 })
 ```
 
+### Generate PKCE pair
+
+`generatePkcePair([length])`
+
+```typescript
+const { codeChallenge, codeVerifier } = generatePckePair()
+```
+
 ### Get Authorization URL
 
-`client.authorizationUrl(state, scope, [nonce], [redirectUri])`
+`async client.authorizationUrl(parameters)`
+
+- parameters: `<Object>`
+  - codeChallenge: `<string>`
+  - state: `<string>` (Optional)
+  - scope: `<string | string[]>` (Optional) **Default**: `myinfo.name openid`
+  - nonce: `<string> | <null>` (Optional) **Default**: Randomly generates a nonce if unspecified
+  - redirectUri: `<string>` (Optional) **Default**: Utilizes the `redirectUri` provided in the constructor
 
 ```typescript
-const { url } = client.authorizationUrl(
-  'state',
-  ['openid', 'myinfo.nric_number'], // or space-concatenated string
-  null, // defaults to randomly generated nonce if unspecified
-  'http://localhost:3000/other_callback', // overrides redirect uri
-)
+const { url } = client.authorizationUrl({
+  codeChallenge: 'zaqUHoBV3rnhBF2g0Gkz1qkpEZXHqi2OrPK1DqRi-Lk', // generated from the previous step
+  state: 'state', // optional as PKCE protects against CSRF
+  scope: ['openid', 'myinfo.name'], // or space-concatenated string
+  nonce: null, // defaults to randomly generated nonce if unspecified
```

Comment:
do we want to encourage users to use the nonce? p sure most users won't if we put this in the README

Reply:
<img width="755" alt="image" src="https://github.com/opengovsg/sgid-client/assets/63008910/7e95ebd8-40a6-43ce-9d06-36bb7eedf3a9">

https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#section-4.7.1

Seems like nonce is encouraged but not mandatory (just like state) because of PKCE protection. i think let's still encourage it nonetheless. i'll change the phrasing to be more similar to state.

Reply:
i've added a comment for nonce in `callback` to specify that it should be the nonce generated from `authorizationUrl`